option.go

```diff
@@ -0,0 +1,244 @@
+		mods = append(mods, dhcpv4.WithDomainSearchList(d.DomainSearch...))
+	}
+	if len(d.NTPServers) > 0 {
+		mods = append(mods, dhcpv4.WithOption(dhcpv4.OptNTPServers(d.NTPServers...)))
+	}
+	if !d.BroadcastAddress.IsZero() {
+		mods = append(mods, dhcpv4.WithGeneric(dhcpv4.OptionBroadcastAddress, d.BroadcastAddress.IPAddr().IP))
+	}
+	if d.DomainName != "" {
+		mods = append(mods, dhcpv4.WithGeneric(dhcpv4.OptionDomainName, []byte(d.DomainName)))
+	}
+	if d.Hostname != "" {
+		mods = append(mods, dhcpv4.WithGeneric(dhcpv4.OptionHostName, []byte(d.Hostname)))
+	}
+	if len(d.SubnetMask) > 0 {
+		mods = append(mods, dhcpv4.WithNetmask(d.SubnetMask))
+	}
+	if !d.DefaultGateway.IsZero() {
+		mods = append(mods, dhcpv4.WithRouter(d.DefaultGateway.IPAddr().IP))
+	}
+
+	return mods
+}
+
+// setNetworkBootOpts purpose is to sets 3 or 4 values. 2 DHCP headers, option 43 and optionally option (60).
+//
+// DHCP Headers (https://datatracker.ietf.org/doc/html/rfc2131#section-2)
+// 'siaddr': IP address of next bootstrap server. represented below as `.ServerIPAddr`.
+// 'file': Client boot file name. represented below as `.BootFileName`.
+//
+// DHCP option
+// option 60: Class Identifier. https://www.rfc-editor.org/rfc/rfc2132.html#section-9.13
+// option 60 is set if the client's option 60 (Class Identifier) starts with HTTPClient.
+func (s *Server) setNetworkBootOpts(ctx context.Context, m *dhcpv4.DHCPv4, n *data.Netboot) func(d *dhcpv4.DHCPv4) {
+	// m is a received DHCPv4 packet.
+	// d is the reply packet we are building.
+	withNetboot := func(d *dhcpv4.DHCPv4) {
+		var opt60 string
+		// if the client sends opt 60 with HTTPClient then we need to response with opt 60
+		if val := m.Options.Get(dhcpv4.OptionClassIdentifier); val != nil {
+			if strings.HasPrefix(string(val), httpClient.String()) {
+				d.UpdateOption(dhcpv4.OptGeneric(dhcpv4.OptionClassIdentifier, []byte(httpClient)))
+				opt60 = httpClient.String()
+			}
+		}
+		d.BootFileName = "/netboot-not-allowed"
+		d.ServerIPAddr = net.IPv4(0, 0, 0, 0)
+		if n.AllowNetboot {
+			a := arch(m)
+			bin, found := ArchToBootFile[a]
+			if !found {
+				s.Log.Error(fmt.Errorf("unable to find bootfile for arch"), "arch", a)
+				return
+			}
+			mac := m.ClientHWAddr
+			uClass := UserClass(string(m.GetOneOption(dhcpv4.OptionUserClassInformation)))
+			ipxeScript := s.IPXEScriptURL
+			if n.IpxeScriptURL != nil {
+				ipxeScript = n.IpxeScriptURL
+			}
+			d.BootFileName, d.ServerIPAddr = s.bootfileAndNextServer(ctx, mac, uClass, opt60, bin, s.IPXEBinServerTFTP, s.IPXEBinServerHTTP, ipxeScript)
+			pxe := dhcpv4.Options{
+				// PXE Boot Server Discovery Control - bypass, just boot from filename.
+				6:  []byte{8}, // or []byte{8}
+				69: binaryTpFromContext(ctx),
+			}
+			d.UpdateOption(dhcpv4.OptGeneric(dhcpv4.OptionVendorSpecificInformation, pxe.ToBytes()))
+		}
+	}
+
+	return withNetboot
+}
+
+// bootfileAndNextServer returns the bootfile (string) and next server (net.IP).
```

Comment:
It seems like we're doing some clever encoding of certain information into URLs.
Perhaps some explanation of that in the comment?